packages/toc/src/generators/notebook/toolbar_generator.tsx

```diff
@@ -95,6 +95,7 @@ function toolbar(options: OptionsManager, tracker: INotebookTracker) {
             ) as boolean;
             options.initializeOptions(
               numbering || options.numbering,
+              options.numberingH1,
```

Comment:
Should maybe it possibly be preserved in the notebook metadata as well, so that the setting is remembered when user refreshes the page/re-opens the notebook later? All other toggles work this way. It would require adding:

```
const numberingH1 = tracker.currentWidget.model!.metadata.get(
   'toc-numberingH1'
) as boolean;
```

changing this line to:

```
numberingH1 || options.numberingH1,
```

And updating the setter to also have `this.notebookMetadata = ['toc-numberingH1', this._numberingH1];` as do other setters:

https://github.com/jupyterlab/jupyterlab/blob/dbdefed9db9332381fe4104bdf53ec314451951f/packages/toc/src/generators/notebook/options_manager.ts#L109-L113